rasa_core/tracker_store.py

```diff
@@ -250,3 +256,137 @@ def retrieve(self, sender_id):
+
+        metadata.create_all(self.engine)
+
+    def keys(self) -> List:
+        """Returns the keys of the items stored in the database"""
+        return self.SQLEvent.__table__.columns.keys()
+
+    def retrieve(self, sender_id: Text) -> DialogueStateTracker:
+        """Creates a tracker from all previously stored events"""
+
+        subquery = self.session.query(self.SQLEvent)
+        query = subquery.filter_by(sender_id=sender_id).all()
+        events = [json.loads(event.data) for event in query]
+
+        if self.domain and len(events) > 0:
+            logger.debug('Recreating tracker '
+                         'from {} stored events'.format(len(events)))
+
+            tracker = DialogueStateTracker.from_dict(sender_id, events,
+                                                     self.domain.slots)
+        else:
+            logger.warning("Can't retrieve tracker from SQL storage.  "
+                           "Returning `None` instead.")
+            tracker = None
+        return tracker
+
+    def save(self, tracker: DialogueStateTracker) -> None:
+        """Updates database with events from the current conversation"""
+
+        if self.event_broker:
+            self.stream_events(tracker)
+
+        events = self._event_buffer(tracker)  # only store recent events
+
+        for event in events:
+            try:
+                data = event.as_dict()
+            except AttributeError as e:
+                logger.warning("Unable to serialise event: {}.  Using "
+                               "__dict__ method instead".format(e))
+                data = event.__dict__
+
+            intent = data.get("parse_data", {}).get("intent", {}).get("name")
+            action = data.get("name")
+            timestamp = data.get("timestamp")
+
+            self.session.add(self.SQLEvent(sender_id=tracker.sender_id,
+                                           type_name=event.type_name,
+                                           timestamp=timestamp,
+                                           intent_name=intent,
+                                           action_name=action,
+                                           data=json.dumps(data)))
+        self.session.commit()
+        logger.debug('Tracker stored to database')
+
+    def _event_buffer(self, tracker: DialogueStateTracker) -> Iterator:
+        """Returns events from the tracker which aren't currently stored"""
+
+        from sqlalchemy import func
+        query = self.session.query(func.max(self.SQLEvent.timestamp))
+        max_timestamp = query.filter_by(sender_id=tracker.sender_id).scalar()
+
+        if max_timestamp is None:
+            max_timestamp = 0
+
+        latest_events = []
+
+        for event in reversed(tracker.events):
+            if event.timestamp > max_timestamp:
+                latest_events.append(event)
+            else:
+                break
+
+        logger.debug('Storing {} recent events'.format(len(latest_events)))
```

Comment:
```suggestion
```

Reply:
Why remove this _and_ the other saving debug statement?  Then there are no logs for this process